app/models/framework_response.rb

```diff
@@ -32,7 +32,14 @@ def update_with_flags!(value)
 
       update!(flags: build_flags)
       clear_dependent_values_and_flags!(old_value)
+
+      # lock the state update to avoid race condition on multiple response patches
+      person_escort_record.with_lock do
```

Comment:
when attempting to calculate the state, a race condition is occurring due to multiple response submission per page. For instance if two responses are sent at the same time, the state is calculated as in progress although it is now completed due to each response being seen as missing at the current time of calculation. Adding a lock prevents this from happening and waits for the first status to be calculated, then the second making sure all response values have been updated.